javascript/sdk/corvy.js

```diff
@@ -1,251 +1,354 @@
-      this._log(`Get user by id failed: ${err.message}`);
-    }
-  }
-
-  async getUserByUsername(username) {
-    try {
-      this._log(`Fetching user by username: ${username}`);
-      const response = await this.client.get(`/users/by-username/${username}`);
-
-      return response.data;
-    } catch (err) {
-      this.emit("error", new Error("Get user by username failed: " + err.message));
-      this._log(`Get user by username failed: ${err.message}`);
-    }
-  }
-
-  async getFlockById(flockId) {
-    try {
-      this._log(`Fetching flock by id: ${flockId}`);
-      const response = await this.client.get(`/flocks/${flockId}`);
-
-      return response.data;
-    } catch (err) {
-      this.emit("error", new Error("Get flock by id failed: " + err.message));
-      this._log(`Get flock by id failed: ${err.message}`);
-    }
-  }
-
-  // -------------------------------
-  // Stuff?
-  // -------------------------------
-
-  get uptime() {
-    return this.startTime ? Date.now() - this.startTime : 0;
-  }
-
-  get formattedUptime() {
-    const ms = this.uptime;
-    const sec = Math.floor((ms / 1000) % 60);
-    const min = Math.floor((ms / (1000 * 60)) % 60);
-    const hr = Math.floor((ms / (1000 * 60 * 60)) % 24);
-    const days = Math.floor(ms / (1000 * 60 * 60 * 24));
-    const parts = [];
-    
-    if (days) parts.push(`${days}d`);
-    if (hr) parts.push(`${hr}h`);
-    if (min) parts.push(`${min}m`);
-    if (sec || parts.length === 0) parts.push(`${sec}s`);
-
-    return parts.join(" ");
-  }
-};
\ No newline at end of file
+        this._log(`📨 Message in nest ${nestId} from ${sender}: '${msgContent}'`);
+        this.emit('messageRaw', message);
+
+        if (msgContent.startsWith(this.prefix)) {
+            const parts = msgContent.slice(this.prefix.length).trim().split(/\s+/);
+            const commandName = parts[0].toLowerCase();
+            const args = parts.slice(1);
+            const argsText = msgContent.slice(this.prefix.length + commandName.length).trim();
+
+            const handled = await this._executeCommand(message, commandName, args, argsText);
+
+            if (!handled) {
+                this.emit('message', message);
+            }
+        } else {
+            this.emit('message', message);
+        }
+    }
+
+    async _executeCommand(message, commandName, args, argsText) {
+        const commandHandler = this.commands.get(commandName);
+
+        if (commandHandler) {
+            let flockId = message.flock_id || this.flockNestMap[message.nest_id.toString()];
+
+            if (!flockId) {
+                await this._fetchFlockForNest(message.nest_id.toString());
+                flockId = this.flockNestMap[message.nest_id.toString()];
+
+                if (!flockId) {
+                    this._log(`⚠️ Could not find flock ID for nest ${message.nest_id}, can't respond`);
+                    return false;
+                }
+
+                message.flock_id = flockId;
+            }
```

Comment:
i think the api should always provide a flock id so you shouldnt need this fallback

Reply:
Commented it out.

Reply:
you only commented out the inner loop when you shouldn't need to use the nestmap at all [blame github for not showing the first line]